### ISSUE TITLE
[Feat] Update to NPM 5 or use Yarn for .lock file

### ISSUE THREAD

**Issue by luchillo17** [typeorm/typeorm#810]
I've come to notice there's no `.lock` file, which means unless you verbosely lock the versions in `package.json` file, it will install latest depending on the semver rule in `package.json`.

To put an example, this is what i see in the `package.json`:
```json
{
    "devDependencies": {
        "chai": "^3.5.0"
    }
}
```
In this case say you installed when `chai` version `3.5.0` was out, and it works ok, however someone else clones the project later and end's up with `3.6.3` due to the `^` that tells npm and yarn to only look at major version for download, then he realizes it doesn't work, why?

Maybe version `3.6.x` introduced a regression, or a breaking change, anyway the person would be unable to know what is the correct version unless he asks the project owner or something like that.

That's where the `.lock` file comes in, previous to NPM 5 it would be named `npm-shrinkwrap.json`, and in order to generate it you have to make it manually, but starting in NPM 5, a `package-lock.json` is automatically generated on install, and keep up to date.

Now yarn already does that with the `yarn.lock` file to lock and commit to the repo which exact versions the project owner is using.

**TL;DR:** Either update to NPM 5, commit the `lock` file and don't worry about this kind of issue ever, or use yarn instead in which it has been always the default behaviour (don't forget to commit the `yarn.lock` file in this scenario).

**Comment by NoNameProvided** [comment]
I have added a `package-lock.json` in a PR but during review I was asked to remove so I think @pleerock is neither a fan of `package-lock.json` so I don't think will happen.

Also I see no benefit for using yarn as npm@5 is almost as fast as yarn.

**Comment by luchillo17** [comment]
@NoNameProvided Yes, it depends on preference, i like yarn because it downloads in parallel, but if he does update to `npm@5` it would be very beneficial to the other project contributors to be able to install the exact same package versions he's using, so i think he should commit the `lock` file it generates, be it with `npm@5` or yarn.

**Comment by pleerock** [comment]
@NoNameProvided actually Im still using old npm version and was not aware about `package-lock.json` file, thats why I thought its something your local stuff and asked to remove it. I'll update my npm version and will commit lock file today

**Comment by luchillo17** [comment]
Nice, maybe this will help me with https://github.com/typeorm/typeorm/issues/826 with testing not working the same as in CI/CD, i'll try later.
